.github/workflows/ci-short.yml

```diff
@@ -0,0 +1,120 @@
+      image: registry.gitlab.com/pgrete/parthenon/cuda11.3-mpi-hdf5
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'true'
+      - name: cpplint
+        run: python ./tst/style/cpplint.py --counting=detailed --recursive src example tst
+      - name: copyright
+        run: |
+          cmake -DCMAKE_CXX_FLAGS=-Werror -Bbuild-copyright-check
+          cmake --build build-copyright-check -t check-copyright
+      - uses: actions/upload-artifact@v3
+        with:
+          name: configure-log-style
+          path: build-copyright-check/CMakeFiles/CMakeOutput.log
+          retention-days: 3
+
+  unit:
+    strategy:
+      matrix:
+        device: ['cuda', 'host']
+    runs-on: self-hosted
+    container:
+      image: registry.gitlab.com/pgrete/parthenon/cuda11.3-mpi-hdf5
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'true'
+      - name: Configure
+        run: |
+          cmake -B build \
+            -DCMAKE_BUILD_TYPE=Debug \
+            -DMACHINE_VARIANT=${{ matrix.device }}-mpi
+      - name: Build
+        run: cmake --build build
+      - name: Test
+        run: |
+          cd build
+          ctest -LE 'performance|regression'
+      - uses: actions/upload-artifact@v3
+        with:
+          name: configure-log-unit-${{ matrix.device }}
+          path: build/CMakeFiles/CMakeOutput.log
+          retention-days: 3
+
+  integration:
+    strategy:
+      matrix:
+        device: ['cuda', 'host']
+    runs-on: self-hosted
+    container:
+      image: registry.gitlab.com/pgrete/parthenon/cuda11.3-mpi-hdf5
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'true'
+      - name: Configure
+        run: |
+          cmake -B build \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DMACHINE_VARIANT=${{ matrix.device }}-mpi
+      # Test example with "variables" and output
+      - name: advection
+        run: |
+          cmake --build build -t advection-example
+          cd build
+          ctest -R regression_mpi_test:output_hdf5
+      # Test example with swarms
+      - name: particle-leapfrog
+        run: |
+          cmake --build build -t particle-leapfrog
+          cd build
+          ctest -R regression_mpi_test:particle_leapfrog
+
+      # Now testing if there are no hidden memcopies between host and device.
+      # Using a static grid (i.e., not AMR) as additional transfers are expected
+      # during loadbalance and refinement, but not for a static grid. We also need to
+      # turn off sparse, since sparse results in additional transfers.
+      # Also delaying start as there are explicit copies during initialization, e.g.,
+      # when the Variable caches are created.
+      - name: host-device-copy
+        if: ${{ matrix.device }} == 'cuda'
+        run: |
+          cd build
+          nsys profile --delay=5 --duration=5 --stats=true -s none -t cuda example/advection/advection-example \
```

Comment:
The success or failure of this test seems to be sensitive to the hardware it is running on, since `--delay=5` and `--duration=5` may pick up the final step, which seems to involve HtoD and DtoH copies.

Reply:
Correct. This is specifically tuned to the CI machine and would need to be adjusted once we migrate.
I'll add a comment next month (when I do the migration to a machine in Hamburg).